src/components/CardHead.js

```diff
@@ -19,24 +28,74 @@ export default function CardHead({
   handleDelete,
   id,
   pathname,
+  setProfiles,
+  profiles,
 }) {
   const { on, toggle } = useToggle(false)
 
   return (
     <StyledHead onClick={toggle}>
       <StatusStyled>{status}</StatusStyled>
-      {on === false
-        ? pathname === '/profile-pool' && (
-            <DeleteSign onClick={() => handleDelete(id)}>
-              <img src={deleteIcon} alt="trashbox" />
-            </DeleteSign>
-          )
-        : ''}
-      <img src={img} alt="portrait" />
+      {on === false && pathname === '/profile-pool' && (
+        <DeleteButton onClick={() => handleDelete(id)}>
+          <Tooltip title="Delete card" placement="left">
+            <img src={deleteIcon} alt="trashbox" />
+          </Tooltip>
+        </DeleteButton>
+      )}
+      <UploadWrapper>
+        <img src={img} alt="portrait" />
+        {pathname === '/profile-pool' && (
+          <UploadButton htmlFor="imageInput">
+            <input
+              type="file"
+              name="image"
+              id="imageInput"
+              hidden="hidden"
+              onChange={handleUpload}
+            />
+            <Tooltip title="Edit profile picture" placement="right">
+              <img src={uploadIcon} alt="plus-sign" />
+            </Tooltip>
+          </UploadButton>
+        )}
+      </UploadWrapper>
       <h1>{name}</h1>
       <h2>{title}</h2>
     </StyledHead>
   )
+
+  function updateProfileImage(url) {
+    const index = profiles.findIndex(profile => profile.id === id)
+    const updatedProfile = { ...profiles[index], imgUrl: url }
+    setProfiles([
+      ...profiles.slice(0, index),
+      updatedProfile,
+      ...profiles.slice(index + 1),
+    ])
+    saveProfilesToLocal('profiles', profiles)
+  }
+
+  function handleUpload(event) {
+    const image = event.target.files[0]
+    const uploadTask = storage.ref(`user-images/${image.name}`).put(image)
+    uploadTask.on(
```

Comment:
Make a function which handles uploadTask.on()

```
function uploadProfileImage(task, imageName) {
  task.on(
      'state_changed',
      snapshot => {},
      error => {
        console.error(error)
      },
      () => {
        storage
          .ref('user-images')
          .child(imageName)
          .getDownloadURL()
          .then(url => {
            updateProfileImage(url)
          })
      }
    )
}
```

Reply:
not sure to fully understand, maybe we can talk about it this afternoon?